src/visualize/DataArray.cpp

```diff
@@ -72,7 +74,11 @@ void DataArray<TDataType>::WriteAscii(std::ostream& file) const
     file << ">\n";
 
     for (auto data : mData)
-        file << " " << std::to_string(data);
+        // https://stackoverflow.com/questions/14644716/how-to-output-a-character-as-an-integer-through-cout
+        if (typeid(TDataType) == typeid(uint8_t))
```

Comment:
I think you can make that a compile time check instead of the run time check with `typeid`.

Could the following work? If so, I would prefer that.
~~~cpp
#include <type_traits>
...
if (std::is_same<TDataType, uint8_t>::value)
   ...
~~~